server/storage/memory_test.go

```diff
@@ -90,3 +91,135 @@ func TestGetChecksumNotFound(t *testing.T) {
 	require.Error(t, err)
 	require.IsType(t, ErrNotFound{}, err)
 }
+
+func TestMemoryGetChanges(t *testing.T) {
+	s := NewMemStorage()
+
+	// non-int changeID
+	c, err := s.GetChanges("foo", 10, "")
+	require.Error(t, err)
+	require.Len(t, c, 0)
+
+	// add some records
+	require.NoError(t, s.UpdateMany("alpine", []MetaUpdate{
+		{
+			Role:    data.CanonicalTimestampRole,
+			Version: 1,
+			Data:    []byte{},
+		},
+		{
+			Role:    data.CanonicalTimestampRole,
+			Version: 2,
+			Data:    []byte{},
+		},
+		{
+			Role:    data.CanonicalTimestampRole,
+			Version: 3,
+			Data:    []byte{},
+		},
+		{
+			Role:    data.CanonicalTimestampRole,
+			Version: 4,
+			Data:    []byte{},
+		},
+	}))
+	require.NoError(t, s.UpdateMany("busybox", []MetaUpdate{
```

Comment:
Also, maybe a `s.UpdateCurrent` with a timestamp update, and `s.UpdateCurrent` with a snapshot update?  To ensure the timestamp update creates a change log, but the snapshot update does not.